### PR TITLE
Instrument registration flow

### PR DESCRIPTION
**Pull request by jimchamp**
In support of #8863

1. Update our `Sentry` objects to allow for capturing handled exceptions
2. Creates new `OLAuthenticationError`, which are now thrown on IA account creation failures
3. Publishes `/account/create` `POST` handler to Sentry

### Testing

1. Deploy this to `testing`
2. Trigger an error during account registration
3. Check Sentry for an `OLAuthenticationError`

### CODE REVIEW

File: openlibrary/accounts/model.py [hunk 1]
```diff
@@ -669,13 +673,13 @@ def create(
                 return ia_account
 
             elif 'screenname' not in response.get('values', {}):
-                errors = '_'.join(response.get('values', {}))
-                raise ValueError(errors)
+                raise OLAuthenticationError('undefined_error')
```
Comment (jimchamp) [inline openlibrary/accounts/model.py:676]: Created new `undefined_error` for this case.  I am not familiar with the types of errors that can occur here, I was unable to find any existing errors in Sentry for insight.

File: openlibrary/accounts/model.py [hunk 2]
```diff
@@ -669,13 +673,13 @@ def create(
...
 
             elif attempt >= retries:
-                ve = ValueError('username_registered')
-                ve.value = _screenname
-                raise ve
+                # error_key: max_retries_exceeded (?)
```
Comment (jimchamp) [inline openlibrary/accounts/model.py:679]:
```suggestion
```

File: openlibrary/accounts/model.py [hunk 3]
```diff
@@ -669,13 +673,12 @@ def create(
...
-                errors = '_'.join(response.get('values', {}))
-                raise ValueError(errors)
+                raise OLAuthenticationError('undefined_error')
 
             elif attempt >= retries:
-                ve = ValueError('username_registered')
-                ve.value = _screenname
-                raise ve
+                e = OLAuthenticationError('username_registered')
+                e.value = _screenname
```
Comment (jimchamp) [inline openlibrary/accounts/model.py:680]: This doesn't seem to be accessed anywhere.

File: openlibrary/plugins/upstream/account.py [hunk 4]
```diff
@@ -316,8 +319,10 @@ def POST(self):
                 return render['account/verify'](
                     username=f.username.value, email=f.email.value
                 )
-            except ValueError:
-                f.note = get_login_error('max_retries_exceeded')
+            except OLAuthenticationError as e:
+                f.note = get_login_error(e.__str__())
```
Comment (jimchamp) [inline openlibrary/plugins/upstream/account.py:323]: `e.__str__()` returns the exception's arguments.  In this case, its the exception's `message`, which is a key for the `LOGIN_ERRORS` dict.

File: openlibrary/plugins/upstream/account.py [hunk 5]
```diff
@@ -316,8 +319,10 @@ def POST(self):
...
+                if sentry.enabled:
+                    sentry.capture_exception(e)
```
Comment (cdrini) [inline openlibrary/plugins/upstream/account.py:325]: We shouldn't need to call this manually; sentry under the hood should be capturing all these errors, even if they're `except`ed. You should be able to test this by hooking up your local environment to sentry.

**Review by mekarpeles — APPROVED** [review]
LGTM, looks like there may be one question about 
```
if sentry.enabled:
                    sentry.capture_exception(e)
```
but maybe we can remove these two lines and merge... and if we need to add them back in, we do 🤷

**Comment by jimchamp** [comment]
https://github.com/internetarchive/openlibrary/labels/State%3A%20Blocked until the exception test handler commit has been removed.

**Pull request merged**